### DISCUSSION TITLE
Override Environment for Testing a Request in Unit Tests

### DISCUSSION THREAD

**Discussion by ceagan** [Kong/insomnia#4626] [Ideas]
If there was a way to provide an Environment JSON Object within a unit test to override other environment values, then it would be possible to create Unit Tests that reuse the same request, altering the call slightly by altering the environment variables referenced in the Request definition. This would make Requests much more useful from the perspective of Unit Tests and reduce the number of Requests required to build out a set of Unit Tests.

Consider a Request stored within a folder that defines the following environment for all Requests in that folder.

```JavaScript
{
  "date": "2000-01-01",
  "amount": "$20"
}
```

A Request in that folder might have a payload that references the date and amount in the environment like this:

```JavaScript
{
  "purchasedate": "{{ ._date }}"
  "purchaseprice": "{{ ._amount }}"
}
```

The desired feature would enable the unit test to override the date variable in the environment for the duration of the unit test by doing something like this:

```JavaScript
insomnia.setTestEnvironment(
{
  "date": "2022-10-10"
}
);

const response1 = await(insomnia.send();
expect(response1.status).to.equal(200);
```

In the above example, the unit test would run with the `date = "2022-10-10"` and `amount = "$20"`.

**Comment by eddex** [comment]
Yesterday I added idea https://github.com/Kong/insomnia/discussions/4975 to add support for overriding the GraphQL query variables in unit tests. But this proposal would make this obsolete since we could simply use environment variables as query variables and then override them.
